tutorials/serverless-grafana-with-iap/index.md

```diff
@@ -0,0 +1,156 @@
+
+To access your Grafana dashboard, Cloud Load Balancer is configured to service HTTPS traffic from CloudRun using a Serverless Network Endpoint Group (NEG) as backend service. Identity-Aware Proxy (IAP) is integrated with the Load Balancer backend service. Client ID and secret are passed to the Load Balancer configuration. IAP provides headers containing the authorization information to applications secured with it ([link to documentation](https://cloud.google.com/iap/docs/signed-headers-howto)). Grafana provides the [functionality](https://grafana.com/docs/grafana/latest/auth/auth-proxy/) to receive exactly such header information for authentication. 
+
+
+### Before you begin
+
+You should have a Google Cloud Platform account and project setup, billing configured for your project, and Terraform installed and enabled.
+  1. [Create or select a Google Cloud project.](https://console.cloud.google.com/project)
+  1. [Enable billing for your project.](https://support.google.com/cloud/answer/6293499#enable-billing)
+  1. Choose a region to host your project in, ideally one that’s close to you. You can find an overview of available regions [here](https://cloud.google.com/compute/docs/regions-zones).
+  1. Make sure you know the domain name where you want to host your Grafana dashboard and are able to edit the A record for this domain.
+
+
+### Configure the OAuth consent screen
+
+Configure an OAuth consent screen for Identity-Aware Proxy. 
+  1. Go to GCP Console > Security > Identity-Aware Proxy. 
+  2. If you didn’t configure a consent screen before, there will be a red warning message prompting you to configure one. 
+  ![Consent missing error message](./iap-consent-not-configured.png)
+  Depending on your setup you can either enable the consent screen for internal or external users. Internal allows only users that are part of your Cloud Identity organisation to use Grafana, but requires you to have a Cloud Identity organisation. You can alternativley select external, which allows you to also add users that are outside your organisation. As long as your app wasn't publish, you will need to whitelist "testers" that can access your application (see optional step 4).
```

Comment:
As per the comment below, seems like creating IAP via TF doesn't work for "external" apps. So can change this text (and point 4)